cli/common/common.go

```diff
@@ -73,16 +102,33 @@ func GetBoard(firmwareIndex *firmwareindex.Index, fqbn string) *firmwareindex.In
 	return board
 }
 
-// GetUploadToolDir is an helper function that downloads the correct tool to flash a board,
+// DownloadRequiredToolsForBoard is an helper function that downloads the correct tool to flash a board,
 // it returns the path of the downloaded tool
-func GetUploadToolDir(packageIndex *packageindex.Index, board *firmwareindex.IndexBoard) *paths.Path {
-	toolRelease := indexes.GetToolRelease(packageIndex, board.Uploader)
+func DownloadRequiredToolsForBoard(pm *packagemanager.PackageManager, board *firmwareindex.IndexBoard) *paths.Path {
```

Comment:
shouldn't this function be placed in `index/download/download.go`?

Reply:
Actually no, because `DownloadRequiredToolsForBoard` it's a convenience wrapper around `download.DownloadTool` (that is performing the download).
Basically `DownloadRequiredToolsForBoard` initialize all the objects and do `feedback.Fatal` in case of errors, more or less the exact purpose of `InitIndexes`.